### PR TITLE
Community library ExternalInterface hooks

### PR DESCRIPTION
**Pull request by idreyn**
A few tweaks to support changes to scratch-flash-online for community library.

### CODE REVIEW

File: src/util/Server.as [hunk 1]
```diff
@@ -39,6 +39,8 @@ import flash.geom.Matrix;
 import flash.net.SharedObject;
 import flash.net.URLLoader;
 
+import scratch.ScratchSprite;
```
Comment (nathan) [inline src/util/Server.as]: Unused import?

File: src/util/ProjectIO.as [hunk 2]
```diff
@@ -508,4 +508,12 @@ public class ProjectIO {
 		recordedAssets[md5] = id;
 		return id;
 	}
+
+	//----------------------------
+	// Community library upload
+	//----------------------------
+
+	public function uploadToCommunityLibrary(spr:ScratchSprite,whenDone:Function):void {
```
Comment (nathan) [inline src/util/ProjectIO.as:544]: Missing a space between `,` and `whenDone`

File: src/scratch/ScratchSprite.as [hunk 3]
```diff
@@ -546,6 +551,19 @@ public class ScratchSprite extends ScratchObj {
 		file.save(zipData, defaultName);
 	}
 
+	private function submitToCommunityLibrary():void {
+		Scratch.app.createProjectIO().uploadToCommunityLibrary(this,function(res:*):void {
+			if(res.okay) {
+			   ExternalInterface.call('CommunityLibrary.submit',res.url);
```
Comment (nathan) [inline src/scratch/ScratchSprite.as:568]: Missing a space between `,` and `res`

File: src/scratch/ScratchSprite.as [hunk 4]
```diff
@@ -546,6 +551,19 @@ public class ScratchSprite extends ScratchObj {
 		file.save(zipData, defaultName);
 	}
 
+	private function submitToCommunityLibrary():void {
+		Scratch.app.createProjectIO().uploadToCommunityLibrary(this,function(res:*):void {
```
Comment (nathan) [inline src/scratch/ScratchSprite.as:566]: Missing a space between `,` and `function`

File: src/scratch/ScratchSprite.as [hunk 5]
```diff
@@ -546,6 +551,19 @@ public class ScratchSprite extends ScratchObj {
...
+			if(res.okay) {
+			   ExternalInterface.call('CommunityLibrary.submit',res.url);
+			} else {
+				DialogBox.notify(
+					'Unable to submit sprite',
+					res.error
+				);
```
Comment (nathan) [inline src/scratch/ScratchSprite.as:573]: You should probably leave this call on one line.

File: src/scratch/ScratchSprite.as [hunk 6]
```diff
@@ -456,6 +458,9 @@ public class ScratchSprite extends ScratchObj {
 		m.addItem('delete', deleteSprite);
 		m.addLine();
 		m.addItem('save to local file', saveToLocalFile);
+		if(Scratch.app.jsEnabled && ExternalInterface.available && ExternalInterface.call('CommunityLibrary.useHTMLBrowser')) {
```
Comment (nathan) [inline src/scratch/ScratchSprite.as]: Missing a space between `if` and `(`

File: src/ui/parts/LibraryPart.as [hunk 7]
```diff
@@ -347,8 +346,17 @@ public class LibraryPart extends UIPart {
...
 	}
 
-	private function spriteFromComputer(b:IconButton):void { importSprite(true) }
-	private function spriteFromLibrary(b:IconButton):void { importSprite(false) }
+	private function spriteFromComputer(b:IconButton):void {
+		importSprite(true);
+	}
+
+	private function spriteFromLibrary(b:IconButton):void {
+		if(ExternalInterface.available && ExternalInterface.call('CommunityLibrary.useHTMLBrowser')) {
```
Comment (nathan) [inline src/ui/parts/LibraryPart.as]: Missing a space between `if` and `(`

File: src/scratch/ScratchSprite.as [hunk 8]
```diff
@@ -546,6 +551,19 @@ public class ScratchSprite extends ScratchObj {
 		file.save(zipData, defaultName);
 	}
 
+	private function submitToCommunityLibrary():void {
+		Scratch.app.createProjectIO().uploadToCommunityLibrary(this,function(res:*):void {
+			if(res.okay) {
```
Comment (nathan) [inline src/scratch/ScratchSprite.as:567]: Missing a space between `if` and `(`

**Comment by sclements** [comment]
@idreyn let's talk about the status of this PR sometime!

**Comment by idreyn** [comment]
@sclements For sure! I'm in Taiwan at the moment though and my internet access is sporadic. You could also get in touch with @jorenl, he was actively working on this as of June.

**Pull request closed without merging**